sycl/source/detail/graph_impl.cpp

```diff
@@ -54,20 +54,63 @@ void graph_impl::remove_root(node_ptr n) {
   MSchedule.clear();
 }
 
+// Recursive check if a graph node or its successors contains a given kernel
+// argument.
+//
+// @param arg The kernel argument to check for.
+// @param currentNode The current graph node being checked.
+// @param deps The unique list of dependencies which have been identified for
```

Comment:
nitpick adding "in/out" to these doxy params would be helpful

Reply:
Done!